src/plugins/pagination/pager-target/pager-target.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, Optional } from '@angular/core';
+import { PluginComponent } from '../../plugin.component';
+import { RootService } from 'ng2-qgrid/infrastructure/component/root.service';
+import { Shortcut } from 'ng2-qgrid/core/shortcut/shortcut';
+
+@Component({
+	selector: 'q-grid-pager-target',
+	templateUrl: './pager-target.component.html'
+})
+
+export class PagerTargetComponent extends PluginComponent {
+	constructor(@Optional() root: RootService) {
+		super(root);
+	}
+
+	private value = '';
+	private input: any;
+
+	keyDown(e: KeyboardEvent) {
+		e.preventDefault();
+
+		const code = Shortcut.translate(e);
+		const digit = Number.parseInt(code);
+		const total = this.total();
+		const firstTimeInput = this.value === '';
```

Comment:
hmm why we need to care about firstTime?
just make if `candidate >= 1 && candidate <= total`?
rename candidate to `page`